contrib/pkg/installmanager/installmanager.go

```diff
@@ -414,32 +427,76 @@ func uploadAdminKubeconfig(cd *hivev1.ClusterDeployment, m *InstallManager) (*co
 		return nil, err
 	}
 
+	return kubeconfigSecret, uploadSecretWithRetries(kubeconfigSecret, m)
+}
+
+func uploadAdminPassword(cd *hivev1.ClusterDeployment, m *InstallManager) (*corev1.Secret, error) {
+	m.log.Infoln("uploading admin username/password")
+	fullPath := filepath.Join(m.WorkDir, adminPasswordRelativePath)
+
+	if _, err := os.Stat(fullPath); os.IsNotExist(err) {
+		m.log.WithField("path", fullPath).Error("admin password file does not exist")
+		return nil, err
+	}
+
+	passwordBytes, err := ioutil.ReadFile(fullPath)
+	if err != nil {
+		m.log.WithError(err).WithField("path", fullPath).Error("error reading admin password file")
+		return nil, err
+	}
+
+	// Need to trim trailing newlines from the password
+	password := strings.TrimSpace(string(passwordBytes))
+
+	s := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf(adminPasswordSecretStringTemplate, m.ClusterName),
+			Namespace: m.Namespace,
+		},
+		Data: map[string][]byte{
+			"username": []byte(kubeadminUsername),
+			"password": []byte(password),
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(cd, s, scheme.Scheme); err != nil {
+		m.log.WithError(err).Error("error setting controller reference on kubeconfig secret")
+		return nil, err
+	}
+
+	return s, uploadSecretWithRetries(s, m)
+}
+
+func uploadSecretWithRetries(s *corev1.Secret, m *InstallManager) error {
+
 	secretUploaded := false
+	var err error
 	// try 10 times to save the admin kubeconfig with some sleeps between attempts
 	for i := 1; i < 11; i++ {
-		err = m.DynamicClient.Create(context.Background(), kubeconfigSecret)
+		err = m.DynamicClient.Create(context.Background(), s)
 		if err != nil {
-			m.log.WithError(err).Warningf("error creating admmin kubeconfig secret (attempt %d)", i)
+			m.log.WithError(err).WithField("secretName", s.Name).Warningf("error creating secret (attempt %d)", i)
 			time.Sleep(sleepBetweenRetries)
 			continue
 		}
-		m.log.WithField("secretName", kubeconfigSecret.Name).Info("uploaded admin kubeconfig secret")
+		m.log.WithField("secretName", s.Name).Info("uploaded secret")
 		secretUploaded = true
 		break
 	}
 
 	if !secretUploaded {
-		m.log.WithError(err).Error("failed to save admin kubeconfig secret")
-		return nil, err
+		m.log.WithError(err).WithField("secretName", s.Name).Error("failed to save admin kubeconfig secret")
```

Comment:
it's really just a generic saving of a secret that failed. not necessarily the 'admin kubeconfig'.